app/main/views/services.py

```diff
@@ -314,117 +301,140 @@ def view_service_submission(service_id):
+    if section is None:
+        abort(404)
+
+    posted_data = _get_formatted_section_data(section)
+    posted_data['page_questions'] = get_section_questions(section)
+
+    try:
+        data_api_client.update_draft_service(
+            service_id,
+            posted_data,
+            current_user.email_address)
+    except HTTPError as e:
+        errors_map = _get_section_error_messages(e, draft['lot'])
+        if not posted_data.get('serviceName', None):
+            posted_data['serviceName'] = draft.get('serviceName', '')
+        return render_template(
+            "services/edit_section.html",
+            section=section,
+            service_data=posted_data,
+            service_id=service_id,
+            return_to=".view_service_submission",
+            dashboard=".framework_dashboard",
+            dash_label="Apply to G-Cloud 7",
+            errors=errors_map,
+            **main.config['BASE_TEMPLATE_DATA']
             )
 
-    if (
-            request.args.get("return_to_summary")
-            or not content.get_next_editable_section_id(section)
-    ):
-        return redirect(
-            url_for(".view_service_submission", service_id=service_id))
+    return_to_summary = bool(request.args.get('return_to_summary'))
+    next_section = content.get_next_editable_section_id(section_id)
+
+    if next_section and not return_to_summary:
+        return redirect(url_for(".edit_service_submission", service_id=service_id, section_id=next_section))
     else:
-        return redirect(url_for(".edit_service_submission",
-                                service_id=service_id,
-                                section=content.get_next_editable_section_id(
-                                    section))
-                        )
+        return redirect(url_for(".view_service_submission", service_id=service_id))
 
 
 def _is_service_associated_with_supplier(service):
-
     return service.get('supplierId') == current_user.supplier_id
 
 
 def _is_service_modifiable(service):
-
     return service.get('status') != 'disabled'
 
 
+def _get_error_message(error, message_key, content):
+    validations = [
+        validation for validation in content.get_question(error)['validations']
+        if validation['name'] == message_key]
+
+    if len(validations):
+        return validations[0]['message']
+    else:
+        return 'There was a problem with the answer to this question'
+
+
+def _is_list_type(key):
+    """Return True if a given key is a list type"""
+    if key == 'serviceTypes':
+        return True
+    return new_service_content.get_question(key)['type'] in ['list', 'checkbox', 'pricing']
+
+
+def _is_boolean_type(key):
+    """Return True if a given key is a boolean type"""
+    return new_service_content.get_question(key)['type'] == 'boolean'
+
+
+def _get_formatted_section_data(section):
+    section_data = dict(
+        list(request.form.items()) + list(request.files.items())
```

Comment:
This explicitly adds files to the `section_data`, but since files are neither list nor boolean they don't end up in the return value anyway